File: programovani/ulohy/files/fibonacciho-posloupnost/clen-fibonacci-rekurze.cpp
#include <iostream>
using namespace std;

// Deklarace a inicializace funkci
int fibonaccihoPosloupnost(int n); // Funkce pro najiti N-teho Fibonacciho cisla pomoci rekurze


int main() {

    // Deklarace a inicializace
    int n = 30; // Inicializujeme kolikate cislo posloupnosti hledame

    // Vypiseme n-ty clen
    cout << fibonaccihoPosloupnost(n - 1) << endl;
    
}

// Funkce pro najiti N-teho Fibonacciho cisla pomoci rekurze
int fibonaccihoPosloupnost(int n) {

    // Pokud je jsme u jednicky
    if(n <= 1) {

        // Vratime 1
        return n;

    }

    // V odstatnich pripadech vratime soucet dvou predchozich cisel posloupnosti
    return fibonaccihoPosloupnost(n - 1) + fibonaccihoPosloupnost(n - 2);
    
}

File: programovani/ulohy/files/fibonacciho-posloupnost/clen-fibonacci-rekurze.py
# Funkce pro testovani prvocisla
def fibonaccihoPosloupnost(n):

    # Pokud je jsme u jednicky
    if(n <= 1):

        # Vratime 1
        return n

    # V odstatnich pripadech vratime soucet dvou predchozich cisel posloupnosti
    return fibonaccihoPosloupnost(n - 1) + fibonaccihoPosloupnost(n - 2)


# Deklarace a incializace
n = 30 # Inicializujeme kolikate cislo posloupnosti hledame

# Vypiseme N-ty clen
print(fibonaccihoPosloupnost(n - 1))